Adaptors/MongoDB/src/TaskTable.cs

```diff
@@ -241,7 +241,8 @@ public class TaskTable : ITaskTable
     var taskCollection = taskCollectionProvider_.Get();
 
     // method is final for taskStatus
-    var result = await taskCollection.UpdateManyAsync(model => model.SessionId == sessionId,
+    var result = await taskCollection.UpdateManyAsync(model => model.SessionId == sessionId         && model.Status != TaskStatus.Completed &&
+                                                               model.Status    != TaskStatus.Failed && model.Status != TaskStatus.Canceled,
```

Comment:
Why are you checking that the status is at Canceled? Is it possible to call this function multiple time to apply this status?

Reply:
These are the 3 final status. Checking them like that allows to do it in only one request